packages/grafana-ui/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { Suspense } from 'react';
+import { Portal } from '../Portal/Portal';
+import { css, cx } from 'emotion';
+
+const modalBackdrop = css`
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  z-index: 1040;
+  background-color: #f7f8fa;
+  opacity: 0.8;
+  backdrop-filter: blur(4px);
+`;
+
+interface Props {
+  title: string;
+  tabs: Array<{
+    title: string;
+    content: string | JSX.Element;
+  }>;
+
+  icon?: string;
+  onDismiss?: () => void;
+}
+
+interface State {
+  selectedIndex: number;
+}
+
+export class Modal extends React.PureComponent<Props, State> {
+  state = {
+    selectedIndex: 0,
+  };
+
+  selectTab = (tabIndex: number) => {
+    this.setState({
+      selectedIndex: tabIndex,
+    });
+  };
+
+  handleDismiss = () => {
+    if (this.props.onDismiss) {
+      this.props.onDismiss();
+    }
+  };
+
+  render() {
+    const { title, icon, tabs } = this.props;
+    const { selectedIndex } = this.state;
+
+    return (
+      <Portal>
+        <div className="modal">
```

Comment:
Since you started on this, would it make sense for new modal to use emotion instead of legacy class names. I know those are going to stay with us for some time. But the styles does not change that often, so I think it's fine having duplicated styles rather than relying on sass